### ISSUE TITLE
Unable to understand how to correctly proceed with account re-registration/resume

### ISSUE THREAD

**Issue by ParticleCore** [BelledonneCommunications/linphone-android#2129]
I have been unable to find any clear explanation on how to correctly structure the following logic:

App is opened for the first time, and the user account is logged in successfully using the snippet code below:

````kotlin
val factory = Factory.instance()
factory.createConfigWithFactory(
    "${context.filesDir.absolutePath}/.linphonerc",
    "${context.filesDir.absolutePath}/linphonerc"
)
val core = factory.createCoreWithConfig(coreConfig, context)
val authInfo = factory.createAuthInfo(username, null, password, null, null, domain, null)
val identity = factory.createAddress("sip:$username@$domain")
val address = factory.createAddress("sip:$domain")?.apply {
    transport = transportType
}
val params = core.createAccountParams().apply {
    identityAddress = identity
    serverAddress = address
    isRegisterEnabled = true
    remotePushNotificationAllowed = true
}

core.removeListener(coreListener) // if I don't do this first then it will pile multiple listeners every time
core.addListener(coreListener) // coreListener is defined outside the current scope

val oldAccount = core.accountList.firstOrNull()

if (oldAccount == null) {
    val account = core.createAccount(params)
    core.addAccount(account)
    core.addAuthInfo(authInfo)
    core.defaultAccount = account
} else {
    // is this how it is suppose to be done?
    core.addAccount(oldAccount)
    core.addAuthInfo(authInfo)
    core.defaultAccount = oldAccount
    core.refreshRegisters()
}

core.start()
````

And that same snippet is used for every subsequent app restart/resume after that.

So far all seems to be working, but what I am very confused about is how to proceed when the app is closed and then opened/resumed. I can't help but think that it is not done in the correct way, especially in the comment where it reads `is this how it is suppose to be done?`

Is the above logic also valid for subsequent app resumes/restarts? The account was already registered, I confirmed that it is being returned correctly in `accountList` and all seems to be working correctly by simply adding it and calling at the end `core.start()`.

So far this has not yet have any calls tested because I have not reached that point, but I want to be sure this logic is done correctly.

Basically, is the above correct for subsequent app resumes/restarts? Am I suppose to create the authinfo again and add it to the core? Do I always have to addListener again despite causing duplicate listeners? Do I always have to call `core.start()` whenever this is done?

Hope you understand my confusion here, I tried looking for examples and documentation as well explaining how this part of the process should be done correctly , but there was nothing available that explained this in a clear way.

**Comment by Viish** [comment]
Hi @ParticleCore,

No, this isn't how it's supposed to be done.
If you don't have an account, create it. If you already have it, do nothing.
As long as you have configured a valid path for local configuration file (which you did), accounts are saved and thus will be automatically loaded when the Core is created using this configuration file the next times.

Cheers,

**Comment by ParticleCore** [comment]
@Viish Much appreciated, I had a feeling I wasn't doing it correctly.

So would it be possible to provide a clear example, or point to one available, on how to go about that scenario when an account is already present? Do I still have to add a listener, do I have to `addAccount` + `addAuthInfo` + `defaultAccount =`, should `refreshRegisters` be called, or `core.start()`?

Or by "do nothing" you mean that as soon as I confirm that I have an account already then I literally do absolutely nothing at all? I don't even call `core.createAccountParams()` nor should I call `core.addListener(coreListener)`?

Sorry for the follow up questions, but the "do nothing" is a bit confusing in the above flow.

**Comment by ParticleCore** [comment]
Understood, I believe I had that in place initially because it was adding the same account multiple times and causing multiple listeners to be triggered on update.

One thing I am still confused is how to actually remove the account from the account list.

I tried doing the following:

````kotlin
val account = core.defaultAccount ?: return
val params = account.params
val clonedParams = params.clone().apply {
    isRegisterEnabled = false
}

account.params = clonedParams

core.removeAccount(account)
core.clearAccounts()
core.clearAllAuthInfo()
core.stop()
core.removeListener(coreListener)
````

And after that, when I try to create a brand account again, the `core.accountList` still has an account, meaning I am only able to create an account the very first time. Is there anything that I am missing here when I want to completely remove the current account?

**Comment by ParticleCore** [comment]
I just realized this might be better off as a separate bug report.
